sysdig/resource_sysdig_monitor_notification_channel_ibm_cloud_function_test.go

```diff
@@ -0,0 +1,122 @@
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: monitorNotificationChannelIBMCloudFunctionWebActionWithWishAuthToken(rText()),
+			},
+			{
+				ResourceName:      "sysdig_monitor_notification_channel_ibm_function.sample2",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: monitorNotificationChannelIBMCloudFunctionWebActionWithWishAuthTokenWithCurrentTeam(rText()),
+			},
+			{
+				ResourceName:      "sysdig_monitor_notification_channel_ibm_function.sample3",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: monitorNotificationChannelIBMCloudFunctionWebActionWithCustomData(rText()),
+			},
+			{
+				ResourceName:      "sysdig_monitor_notification_channel_ibm_function.sample4",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: monitorNotificationChannelIBMCloudFunctionCloudFunction(rText()),
+			},
+			{
+				ResourceName:      "sysdig_monitor_notification_channel_ibm_function.sample5",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func monitorNotificationChannelIBMCloudFunctionWebAction(name string) string {
+	return fmt.Sprintf(`
+resource "sysdig_monitor_notification_channel_ibm_function" "sample1" {
+	name = "Example Channel %s - IBM Function"
+	ibm_function_type = "WEB_ACTION"
+	url = "https://eu-gb.functions.cloud.ibm.com/api/v1/web/namespaces/eeeeeeee-623b-4776-ba35-4065bcbfee7b/actions/hello-world/helloworld?param=true"
+	whisk_auth_token = "xxx"
+}`, name)
+}
+
+func monitorNotificationChannelIBMCloudFunctionWebActionWithWishAuthToken(name string) string {
+	return fmt.Sprintf(`
+resource "sysdig_monitor_notification_channel_ibm_function" "sample2" {
+	name = "Example Channel %s - IBM Function"
+	ibm_function_type = "WEB_ACTION"
+	url = "https://eu-gb.functions.cloud.ibm.com/api/v1/web/namespaces/eeeeeeee-623b-4776-ba35-4065bcbfee7b/actions/hello-world/helloworld?param=true"
+	whisk_auth_token = "xxx"
+}`, name)
+}
+
+func monitorNotificationChannelIBMCloudFunctionWebActionWithWishAuthTokenWithCurrentTeam(name string) string {
+	return fmt.Sprintf(`
+resource "sysdig_monitor_notification_channel_ibm_function" "sample3" {
+	name = "Example Channel %s - IBM Function"
+	ibm_function_type = "WEB_ACTION"
+	url = "https://eu-gb.functions.cloud.ibm.com/api/v1/web/namespaces/eeeeeeee-623b-4776-ba35-4065bcbfee7b/actions/hello-world/helloworld?param=true"
+	share_with_current_team = true
+}`, name)
+}
+
+func monitorNotificationChannelIBMCloudFunctionWebActionWithCustomData(name string) string {
+	return fmt.Sprintf(`
+resource "sysdig_monitor_notification_channel_ibm_function" "sample4" {
+	name = "Example Channel %s - IBM Function"
+	ibm_function_type = "WEB_ACTION"
+	url = "https://eu-gb.functions.cloud.ibm.com/api/v1/web/namespaces/eeeeeeee-623b-4776-ba35-4065bcbfee7b/actions/hello-world/helloworld?param=true"
```

Comment:
Do we need to hardcode url like this? If we switch envs, tests will start failing

Reply:
Also this will not work on staging?

Reply:
urls for ibm functions are not validated, so you can put anything with a precise pattern there and it will work